### ISSUE TITLE
Dont Learn to Code  Learn to Automate

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#6259]
Don&rsquo;t Learn to Code &mdash; Learn to Automate<br>
<br>
https://ift.tt/2HHaP8H<br>
<br>
Andy Bailey<br>
<br>
<div>
<div>
<h1>Don&rsquo;t Learn to Code &mdash; Learn to Automate</h1>
<p>Does anyone remember a few years ago, when <a href="http://www.bbc.com/news/technology-16440126">the mayor of New York decided to learn to program</a>? It was a heady time, because it wasn&rsquo;t just him. I remember these surreal commercials where Miami Heat forward Chris Bosh was encouraging children to learn to code for the good of humanity or something. There was this sudden, overwhelming sentiment that humanity should abandon the folly of any non-programming pursuit and learn them some Ruby or whatever. Andy Warhol, were he alive in 2012, no doubt would have said, &ldquo;in the future, everyone will write code for 15 minutes.&rdquo;</p>
<p>Jeff Atwood wrote an interesting rebuttal to this zeitgeist, entitled, &ldquo;<a href="http://blog.codinghorror.com/please-dont-learn-to-code/">Please Don&rsquo;t Learn to Code.</a>&rdquo; The covers a good bit of ground and makes some interesting points, but the overarching thesis seems to be, &ldquo;avoid thinking of writing code as the goal and learn to solve problems.&rdquo; I think this is an excellent, philosophical point, but I&rsquo;d like to add a bit of nuance.</p>
<p>I&rsquo;ve written in the past about how important I think that it is to be a problem solver, to the point where I <a href="https://www.daedtech.com/job-titles-be-like-the-wolf">wrote a post about liking the title &ldquo;problem solver.&rdquo;</a> So please don&rsquo;t think I disagree with his take that a lot of programmers get too hung up with the particulars of code. I don&rsquo;t &mdash; I think that&rsquo;s a very common issue. But, at the same time, I think the mayor of New York and Chris Bosh and others have a point that Jeff doesn&rsquo;t really address, per se. Specifically, the world is getting dramatically more technical, which means that a lot of pursuits are being automated out of existence, while other pursuits require an increasing degree of technical savvy. My fiancee, a professional copy editor, is finding aspects of her job to be easier if she knows a bit of HTML and CSS.</p>
<p>So while I wince alongside Jeff at the thought of people randomly learning programming languages because they think it&rsquo;ll make them rich or because they want to be a person that writes lots of code, I don&rsquo;t think we can simply say, &ldquo;stay out unless you&rsquo;re serious and willing to spend years getting good.&rdquo; The rapidly evolving technical landscape has created this black hole of technical savvy that&rsquo;s sucking in even people well past the event horizon.</p>
<p>The advice that I&rsquo;d offer on this subject creates a pretty fine distinction. I don&rsquo;t think that everyone needs to learn to code by any stretch. What I think that everyone needs to start learning about and understanding is how to <em>automate</em>. Or, if not how to do it themselves, at least how to recognize things that could be automated and have meaningful discussions about whether the effort is worth it or not.</p>
<h3>Automation in Real Life</h3>
<p>It might be best to offer an example here. Tonight, I was working with a friend on some course materials to be communicated via Power Point. We were making slides that were largely images with a few large-font words scattered here and there. We&rsquo;d paste the pictures into the slide deck, resize a little and then use my Mac&rsquo;s trackpad laboriously to center the images. At one point, he said, &ldquo;there should be some shortcut key in Power Point that you can hit and it just auto-centers the thing vertically and horizontally.&rdquo;</p>
<p>That, my friends, is automation at the moment of conception. Centering something on a Power Point is just the kind of brainless, maddening, time-consuming task that makes a good candidate for automation. You know what I&rsquo;m talking about, where you drag the thing around using movements as microscopic as possible, hoping to see that line appear that indicates that it&rsquo;s in the middle. It&rsquo;s so hard to find and you&rsquo;ve got to go so &mdash; wait, there! Crap! Missed it. Back up, try it again.</p>
<p><a rel="attachment wp-att-4452" href="https://www.daedtech.com/rip-rtfm/cryingdev"><img alt="CryingDev">
<div class="feedlyNoScript"><img alt="CryingDev" src="https://www.daedtech.com/wp-content/uploads/2014/11/CryingDev-300x213.jpg"></div>
</a></p>
<p>Of course, the conversation doesn&rsquo;t end here, by any stretch. The talent for which I&rsquo;m advocating &mdash; the talent of savvy automation &mdash; involves an assessment like the following, and a <em>potential</em> automation at the end.</p>
<h4>Does Power Point already provide this capability and we just don&rsquo;t know about it?</h4>
<p>Doesn&rsquo;t seem like it after a bit of quick googling. If it did, a <a href="https://pptcrafter.wordpress.com/2013/05/15/powerpoint-secrets-alignment/">blog about Power Point secrets</a> would probably just need two lines to describe it &ldquo;Step 1: hit magic key sequence. Step 2: there is no step 2.&rdquo; But still, it might exist.</p>
<h4>Is it worth doing more research?</h4>
<p>Not right now. We&rsquo;re almost done with the slide deck. On a broader scale, it might be worth doing.</p>
<h4>Would it be worth it to us over the long haul?</h4>
<p>Hard to say. I&rsquo;m not entirely sure how much time in my life is wasted to this problem. It&rsquo;s probably some but I&rsquo;m not the most avid Power Point user, so you figure it might take years for it to add up to hours of my time. And who knows how long fixing this problem&nbsp;would take?</p>
<h4>Would it be worth it <em>for humanity</em> over the long haul?</h4>
<p>Hmmm&hellip; almost certainly. Even if I waste hours on this only after years of Power Point use, <em>humanity</em> probably wastes hours on this after microseconds go by. Lots of people somewhere are swearing at Power Point right this very instant.</p>
<h4>Could we sell it if we made it?</h4>
<p>Meh, doubtful. But maybe for a pittance here and there. Or maybe we could just do it, donate it to the greater good, and reap a reputation benefit.</p>
<h4>Okay, so what&rsquo;s the next action?</h4>
<p>Let&rsquo;s set aside a time box of an hour to research this at a good time for low priority tasks, such as sitting in an airport. A nice interim goal might be to see if we can get it going in a limited scenario. On just a blank slide, does the &ldquo;ctrl-E&rdquo; shortcut do the job for horizontal centering? If so, can we find a way to do vertical centering&nbsp;and then somehow chain those shortcuts? If yes, then, bam, problem solved with just a bit of configuration. We&rsquo;ll know we&rsquo;ve succeeded when we can hit a single shortcut and have horizontal and vertical center of a single image on a blank slide. If that little test works, we can write a blog post, bask in some win, and then move on to more scenarios.</p>
<h3>Poor Man&rsquo;s Coding?</h3>
<p>Okay, so that was a little hokey, but I&rsquo;d like to point some things out here. First up was identification of a crappy task and the recognition of an automation candidate. Figuring out the exact coordinates for centering something is the perfect job for a computer. After that came a sequence of questions contemplating the possibility that there might be an existing solution, that coming up with a new one might not be worth it, or that coming up with a new one might be time consuming enough as to offset any gains. In the world of software development, that goes by various names such as &ldquo;discovery,&rdquo; &ldquo;requirements analysis,&rdquo; and &ldquo;sprint 0.&rdquo;</p>
<p>After that came a tentative plan of action along with some risk mitigation. Let&rsquo;s invest a bit of effort in seeing what we can do, but let&rsquo;s cap that amount of effort so that we don&rsquo;t run off tilting at windmills. And then there was a concrete strategy that involved carving a larger potential effort down into the smallest slice that might provide some incremental value. This is known in the software development biz as an agile (or, if you want, &ldquo;lean&rdquo;) approach to a software project. And then, finally, in spite of the lack of writing of actual code, came implementation and a clear, verifiable description of what success looks like. In the biz, that&rsquo;s called &ldquo;Acceptance Test Driven Development&rdquo; (ATDD).</p>
<p>But forget the terms and the parallels to programmer shop-talk. The more important point is that successful software development projects &mdash; projects that involve code and IDEs and compilers and whatnot &mdash; are just a special case of successful automation projects. You can automate all manner of things, even some very non-trivial things, without actually writing any code. In this vein, Jeff&rsquo;s point is absolutely spot on. Coders like to code, but writing code ought not to be a first class goal when there are problems to be solved.</p>
<p><a rel="attachment wp-att-2791" href="https://www.daedtech.com/what-your-job-descriptions-are-saying-to-developers-about-your-company/workharder"><img alt="WorkHarder">
<div class="feedlyNoScript"><img alt="WorkHarder" src="https://www.daedtech.com/wp-content/uploads/2013/03/WorkHarder.jpg"></div>
</a></p>
<p>Throughout human history, there&rsquo;s been a sort of &ldquo;pain is gain&rdquo; approach to the repetitive. There was value in putting your head down, getting into a rhythm, and working hard at menial tasks. But throughout most of that human history, we didn&rsquo;t have computers. It turns out that computers are really, really good at doing repetitive, menial tasks &mdash; tasks that involve precision and not judgment. They&rsquo;re far better at them than we are, so it makes sense to let computers do them.</p>
<p>It&rsquo;s obtuse to suppose that a prerequisite for every job in the future will be the ability to implement sophisticated, specialized computer applications. But it&rsquo;s not at all obtuse to suppose that, given the ubiquity of computing, a prerequisite for every job in the future will be the ability to recognize which tasks are better suited for humans and which for computers. Learn at least to recognize which parts of your job are a poor use of your time. After that, perhaps learn to use your ingenuity and creativity to automate using the tools that you know (such as googling for solutions, leveraging apps, etc). And, if you&rsquo;ve come that far, maybe it&rsquo;s time to roll up your sleeves and take the plunge into learning to code a little bit to help you along. Because, while there&rsquo;s no need&nbsp;for the mayor of New York to write any code, it couldn&rsquo;t hurt him to be ready to jump if something opens up in the city&rsquo;s IT department.</p>
</div>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via DaedTech<br>
<br>
February 14, 2020 at 10:21PM
